### ISSUE TITLE
File is not published when redirect JavaScript output to directory

### ISSUE THREAD

**Issue by msmolka** [Microsoft/TypeScript#7048]
I'm using Visual Studio 2015 (TypeScript 1.7.6), and I've set option for release:
Redirect JavaScript output to directory:  obj\Release\JsGen

File is generated in this folder, however it is not copied to proper folder when I'm publishing web site.
File is only copied when nothing is set in this option.

**Comment by mhegazy** [comment]
We will need a project file to be able to debug this further. also can you give the nightly nuget package a try: https://github.com/Microsoft/TypeScript/wiki/Nightly-drops#using-nuget-with-msbuild we have made multiple fixes to this area in the past few weeks.

**Comment by paulvanbrenk** [comment]
@msmolka Further a diagnostic (/v:diag when invoking MSBuild on the command line) build log also would help a lot. (Please, post it on a 'gist' and not inline in the bug.)

**Comment by msmolka** [comment]
@paulvanbrenk   Hello, I've run publish of temp web site using folowing parameters
msbuild WebApp.csproj /v:diag /p:DeployOnBuild=true /p:Configuration=Release /fileLogger  /p:PublishProfile=Test

And published it here
https://gist.github.com/msmolka/4c8b385d6fed8b2d9060#file-msbuild-log

**Comment by paulvanbrenk** [comment]
@msmolka What is the proper folder it should be copied too?

I see the `.js` and `.js.map` file being copied here:

`  Copying obj\Release\Package\PackageTmp\obj\Release\JsGen\test.js to D:\Temp\Test\web\_Publish\obj\Release\JsGen\test.js.
  Copying obj\Release\Package\PackageTmp\obj\Release\JsGen\test.js.map to D:\Temp\Test\web\_Publish\obj\Release\JsGen\test.js.map.`

**Comment by msmolka** [comment]
@paulvanbrenk Hello, I'm new in TypeScript, and I think I did not understood the out directory settings. My understanding was to generate js files as temporary file in defined directory, and on Publish that temp files should be copied back to the pace where original TypeScript file is, not to related out directory. So if I set obj, the file is temporary compiled there then copied to the place of original ts file during publish. 
So, my understanding was wrong. I think we can close this one.
